includes/class-sensei-settings-api.php

```diff
@@ -931,6 +931,12 @@ public function validate_fields( $input ) {
 		$options = $this->get_settings();
 
 		foreach ( $this->fields as $k => $v ) {
+			if ( 'color' === $v['type'] ) {
+				$input[ $k ] = str_replace( '#', '', $input[ $k ] );
+				if ( ! ctype_xdigit( $input ) && strlen( $input[ $k ] ) !== 6 ) {
```

Comment:
The `ctype_xdigit( $input )` will always be `false`, because `$input` is an array, not a string or number. Which means `! ctype_xdigit( $input )` will always be `true`.

So this line only checks for the input length. It doesn't prevent the user from entering invalid hex color. For example I was able to enter `nonhex` as a value and it accepts it as correct color. 😄 
```suggestion
				if ( ! ctype_xdigit( $input[ $k ] ) || strlen( $input[ $k ] ) !== 6 ) {
```

Reply:
Thanks @dadish !!